src/ucs/sys/sys.c

```diff
@@ -541,7 +541,11 @@ ucs_status_t ucs_sysv_alloc(size_t *size, void **address_p, int flags, int *shmi
     }
 
     /* Attach segment */
-    ptr = shmat(*shmid, NULL, 0);
+    if (*address_p) {
```

Comment:
Indeed, I don't like then one argument depends on other one inside a single call. In this case, here is no some difficult calculations, but from my opinion, this is a bad practice. What do you think?